components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,450 @@
+			end
+		end
+	end
+
+	return self.prizes
+end
+
+function PrizePool:_readPlacements(args)
+	local currentPlace = 0
+	for placementIndex = 1, math.huge do
+		if not args[placementIndex] then
+			break
+		end
+
+		local placementInput = Json.parseIfString(args[placementIndex])
+		local placement = Placement(placementInput, currentPlace + 1, self.opponentType)
+
+		currentPlace = placement.placeEnd
+		table.insert(self.placements, placement)
+	end
+
+	return self.placements
+end
+
+function PrizePool:setConfig(option, value)
+	self.options[option] = value
+	return self
+end
+
+function PrizePool:addCustomConfig(name, default, func)
+	config[name] = {
+		default = default,
+		func = func
+	}
+	return self
+end
+
+-- TODO extract parts of this to the future Enum class
+local CUSTOM_TYPES_OFFSET = 100
+local CUSTOM_TYPES_USED = 0
+function PrizePool:addCustomPrizeType(enum, data)
+	prizeTypes[enum] = CUSTOM_TYPES_OFFSET + CUSTOM_TYPES_USED
+	CUSTOM_TYPES_USED = CUSTOM_TYPES_USED + 1
+	prizeData[prizeTypes[enum]] = data
+	return self
+end
+
+function PrizePool:addPrize(enum, index, data)
+	assert(prizeTypes[enum], 'addPrize: Not a valid prizeEnum!')
+	assert(Logic.isNumeric(index), 'addPrize: Index is not numeric!')
+	table.insert(self.prizes, {id = enum .. index, enum = prizeTypes[enum], index = index, data = data})
+	return self
+end
+
+function PrizePool:setWidgetInjector(widgetInjector)
+	assert(widgetInjector:is_a(WidgetInjector), "setWidgetInjector: Not a WidgetIjector")
+	self.widgetInjector = widgetInjector
+	return self
+end
+
+function PrizePool:setLpdbInjector(lpdbInjector)
+	-- TODO: Add type check
+	self.lpdbInjector = lpdbInjector
+	return self
+end
+
+function PrizePool:_hasUsdPrizePool()
+	return (Table.any(self.placements, function (_, placement)
+		return placement.hasUSD
+	end)) or (self.options.autoUSD and Table.any(self.prizes, function (_, prize)
+		return prize.enum == prizeTypes.LOCAL_CURRENCY
+	end))
+end
+
+--
+-- Static Functions
+--
+
+--- Remove all non-numeric characters from an input and changes it to a number.
+-- Most commonly used on money inputs, as they often contain , or .
+function PrizePool._parseInteger(input)
+	return tonumber((string.gsub(input, '[^%d.]', '')))
+end
+
+--- Checks if the input matches the format of a date
+function PrizePool._isValidDateFormat(date)
+	if String.isEmpty(date) then
+		return false
+	end
+	return string.match(date, '%d%d%d%d-%d%d-%d%d') and true or false
+end
+
+--- Fetches the LPDB object of a tournament
+function PrizePool._getTournamentInfo(pageName)
+	return mw.ext.LiquipediaDB.lpdb('tournament', {
+		conditions = '[[pagename::' .. pageName .. ']]',
+		limit = 1,
+	})[1]
+end
+
+--- Get the default date based on wiki-variables set in the Infobox League
+function PrizePool._getTournamentDate()
+	return Variables.varDefaultMulti('tournament_enddate', 'tournament_edate', 'edate', TODAY)
+end
+
+---
+function Placement:init(args, placeStart, opponentType)
```

Comment:
What is a "place"? Is this a placement in LPDB?

Reply:
A `Placement` is the equivalent of `Template:Prize table slot` in the current template iteration (which will be named `Template:Placement` per comments in the RFC), and it has a set of opponents (teams/players).
A `Placement` has a range from `placementStart` to `placementEnd`, for example `5` to `8`, and is expected to have the same amount of opponents as the range allows (`4` is the `5-8` example). The '`placementStart`-`placementEnd`' is stored into the field `place` in `LPDB_placement`.

Reply:
"placement" already refers to the place range (e.g. place=5-8)

the contents of `{{Prize pool slot}}` should have a different name

maybe rename to `Slot`?

Reply:
Using `Template:Slot` for now as `Template:Placement` was used by the AutoPointsTable for bg displays (however that should be dropped most likely, as it's a module that using a none-complete template).
If in the end `Template:Slot` will be used over `Template:Placement` in the wiki-code, I agree that renaming the Class as well to match.